third_party/terraform/resources/resource_compute_instance.go.erb

```diff
@@ -1750,14 +1710,18 @@ func resourceComputeInstanceUpdate(d *schema.ResourceData, meta interface{}) err
 			}
 		}
 
-		for _, updateCall := range updatesToNIWhileStopped {
-			op, err := updateCall()
+		// If the instance stops it can invalidate the fingerprint for network interface.
+		// refresh the instance to get a new fingerprint
+		if len(updatesToNIWhileStopped) > 0 {
+			instance, err = config.clientComputeBeta.Instances.Get(project, zone, instance.Name).Do()
 			if err != nil {
-				return errwrap.Wrapf("Error updating network interface: {{err}}", err)
+				return err
 			}
-			opErr := computeOperationWaitTime(config, op, project, "network interface to update", userAgent, d.Timeout(schema.TimeoutUpdate))
-			if opErr != nil {
-				return opErr
+		}
+		for _, patch := range updatesToNIWhileStopped {
+			err := patch(instance)
+			if err != nil {
```

Comment:
nit: since you don't need the err later and it's the only return value from patch, you could consolidate these two lines with:
```suggestion
			if err := patch(instance); err != nil {
```

Reply:
I'm not such a huge fan of this strategy as, in my opinion, it detracts from readability. With patch on its own line it's clear that it's an operation. With it consolidated into the if I find it harder to tell what's going on.